pyproject.toml

```diff
@@ -17,6 +17,7 @@ dependencies = [
     "cached-property; python_version < '3.8'",
     "tqdm > 4",
     "jiter>=0.4.0, <1",
+    "argcomplete >= 1.12.0"
```

Comment:
I'd like to avoid adding a new required dependency, can we add this as an optional dependency?

I'm thinking something like this.
```
pip install openai[cli]
```

Reply:
Thank you for the feedback! To avoid making argcomplete a required dependency, we plan to:

1. Change `argcomplete` from a required to an optional dependency in `pyproject.toml`.
2. Add a proxy file for `argcomplete` to handle optional imports.
3. Update the `__init__.py` file in the extras directory to reflect these changes.
4. Modify the README to instruct users to use the `[cli]` option to install `argcomplete`.

Please let us know if there are any additional changes we should consider.

Reply:
That sounds good to me!

For what it's worth I don't think we need the complexity of a proxy here though as the dependency is only ever used in one place anyway we can just change that call site to `try ... except ImportError`

Reply:
We have included the changes for making `argcomplete` an optional dependency, the `try...except ImportError` to the `_cli.py`, as well as the `pip install openai[cli]` command in the README. Let me know if there's anything else that needs modification. Thank you!